src/plugins/filedelete/filedelete.c

```diff
@@ -189,25 +184,15 @@ static void grab_file_by_handle(drakvuf_t drakvuf, uint32_t vcpu_id, vmi_instanc
                                 drakvuf_trap_info_t *info, addr_t handle)
 {
     uint8_t type_index = 0;
-    addr_t thread, process;
-    /*
-     * fs_base/gs_base in the info->regs structure are not actually filled in
-     * by Xen for vm_events, so we need to manually ask for these each time
-     */
-    reg_t fsgs;
-    if(pm == VMI_PM_IA32E)  {
-        vmi_get_vcpureg(vmi, &fsgs, GS_BASE, vcpu_id);
-    } else {
-        vmi_get_vcpureg(vmi, &fsgs, FS_BASE, vcpu_id);
-    }
+    addr_t process;
 
-    vmi_read_addr_va(vmi, fsgs + offsets[KPCR_PRCB] + offsets[KPRCB_CURRENTTHREAD],
-                     0, &thread);
-    vmi_read_addr_va(vmi, thread + offsets[KTHREAD_PROCESS],
-                     0, &process);
+    process=drakvuf_get_current_process(drakvuf,vcpu_id);
```

Comment:
This could be just a single line of `addr_t process = drakvuf_get_current_process(drakvuf,vcpu_id);`